src/table/cells/handle-cell-focus.js

```diff
@@ -11,33 +11,76 @@ export const updateFocus = (rowElements, cellCoord, shouldFocus = true) => {
-  focusedCellCoord.current = [rawRowIdx + 1, rawColIdx];
-  updateFocus(rowElements, [rawRowIdx + 1, rawColIdx]);
+  updateFocus({ rowElements, cellCoord: focusedCellCoord, shouldFocus: false });
+  setfocusedCellCoord([rawRowIdx + 1, rawColIdx]);
+  updateFocus({ rowElements, cellCoord: [rawRowIdx + 1, rawColIdx], shouldFocus: true });
 };
 
-export const handleClickToFocusHead = (columnIndex, focusedCellCoord, rootElement) => {
+export const handleClickToFocusHead = (columnIndex, focusedCellCoord, rootElement, setfocusedCellCoord) => {
   const rowElements = rootElement.getElementsByClassName('sn-table-row');
-  updateFocus(rowElements, focusedCellCoord.current, false);
-  focusedCellCoord.current = [0, columnIndex];
-  updateFocus(rowElements, [0, columnIndex]);
+  updateFocus({ rowElements, cellCoord: focusedCellCoord, shouldFocus: false });
+  setfocusedCellCoord([0, columnIndex]);
+  updateFocus({ rowElements, cellCoord: [0, columnIndex], shouldFocus: true });
 };
 
-export const handleResetFocus = (focusedCellCoord, rootElement, shouldRefocus, hasSelections) => {
+export const handleResetFocus = ({
+  focusedCellCoord,
+  rootElement,
+  shouldRefocus,
+  hasSelections,
+  setfocusedCellCoord,
+}) => {
   const rowElements = rootElement.getElementsByClassName('sn-table-row');
-  updateFocus(rowElements, focusedCellCoord.current, false);
+  updateFocus({ rowElements, cellCoord: focusedCellCoord, shouldFocus: false });
 
   // If we have selections ongoing, we want to stay on the same column
-  const nextcell = [0, hasSelections ? focusedCellCoord.current[1] : 0];
-  focusedCellCoord.current = nextcell;
+  const nextcell = [0, hasSelections ? focusedCellCoord[1] : 0];
+  setfocusedCellCoord(nextcell);
 
   if (shouldRefocus.current) {
-    updateFocus(rowElements, nextcell);
+    updateFocus({ rowElements, cellCoord: nextcell });
     shouldRefocus.current = false;
   } else {
     rowElements[0]?.getElementsByClassName('sn-table-cell')[nextcell[1]].setAttribute('tabIndex', '0');
   }
 };
+
+export const handleNavigateTop = ({ tableSection, focusedCellCoord, rootElement }) => {
+  const MIN_ROW_COUNT = 2;
+  if (!tableSection.current?.scrollTo) return;
+
+  if (focusedCellCoord[0] < MIN_ROW_COUNT) {
+    tableSection.current.scrollTo({
+      top: 0,
+      behavior: 'smooth',
+    });
+  } else {
+    const [x, y] = focusedCellCoord;
+    const rowElements = rootElement.getElementsByClassName('sn-table-row');
+    const cell = rowElements[x]?.getElementsByClassName('sn-table-cell')[y];
+
+    if (cell.offsetTop - cell.offsetHeight * MIN_ROW_COUNT < tableSection.current.scrollTop) {
+      const targetOffsetTop = tableSection.current.scrollTop - cell.offsetHeight;
+      tableSection.current.scrollTo({
+        top: targetOffsetTop <= 0 ? 0 : targetOffsetTop,
+        behavior: 'smooth',
+      });
+    }
+  }
+};
+
+export const handleRestoreFocusOnPageChange = ({ rootElement, nextCoords, setfocusedCellCoord }) => {
```

Comment:
What's the difference between this and `handleResetFocus`?

Reply:
✅ fixed